src/plugins/discover/public/application/index.tsx

```diff
@@ -28,11 +26,9 @@ export const renderApp = ({ element }: AppMountParameters) => {
       iconType: 'glasses',
     });
   }
-  const app = discoverRouter(services, history);
-  ReactDOM.render(app, element);
+  toMountPoint(discoverRouter(services, history))(element);
```

Comment:
I think we should still use the returning method here to unmount the componen later, i.e. `const unmount = toMountPoint()()` and call that where we earlier called the `unmountComponentAtNode` method directly.

Reply:
very true, changed that